File: ruby/0268-missing-number.rb
def missing_number(nums)
  missing = 0
  (0..nums.length).each do |i|
    nbr = (nums[i].nil? ? 0 : nums[i])
    missing += (i - nbr)
  end
  missing
end

#  Solution using Gauss summation
def missing_number(nums)
  gauss = (nums.length / 2.0) * (nums.length + 1)
  nums.each { |n| gauss -= n }
  gauss.to_i
end

File: csharp/0268-missing-number.cs
public class Solution {
    public int MissingNumber(int[] nums) {
        int sum = 0;
        int total = nums.Length * (nums.Length + 1) / 2;
        for (int i = 0; i < nums.Length; i++) {
            sum += nums[i];
        }
        return total - sum;
    }
}